shared/reducers/list.js

```diff
@@ -36,6 +36,13 @@ const list = (state = initialState, action) => {
     return update(state, {
       currentPostedUri: { $set: action.uri }
     });
+  case types.toggleSaved:
+    let eventIndex = state.dataSource.findIndex((post) => {
```

Comment:
You can just write it like
const eventIndex = state.dataSource.dinfIndex(post => post.key === action.eventKey);